### PR TITLE
Ace unload on garage

### LINKED ISSUE #2447: [Feature request]: ACE cargo should be extracted on garaging
I was going to put this in the general vehicle deleted EH, but it turns out that ACE already deletes the attached objects in its own deleted EH, so there's no perf issue. Not sure if that was added recently or I mis-tested before.

Some working code. Not sure if you'd want this added straight to the garage code or a public proxy:
```
{
    if !(_x isEqualType objNull) then { continue };
    if (typeOf _x in ["ACE_Wheel", "ACE_Track"]) then { continue };
    [_x, _veh] call ace_cargo_fnc_unloadItem;
} forEach (_veh getVariable ["ace_cargo_loaded", []]);
```

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [x] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    - added unload for ace cargo on unload, also refunds ltc crates (emptying its contents to arsenal first)
    - fixed fuel containers interaction and important variables not being restored on load

### Please specify which Issue this PR Resolves.
closes #2447 

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
 - load items into ace cargo
 - garage vehicle
 - buy fuel container
 - save and load game
 - garage fuel container
********************************************************
Notes: as purchased light sources cost money we should consider refunding or restoring them on load

### CODE REVIEW

File: A3A/addons/core/functions/Save/fn_loadStat.sqf [hunk 1]
```diff
@@ -28,6 +28,26 @@ Example:
 #include "..\..\script_component.hpp"
 FIX_LINE_NUMBERS()
 
+/*
+    Function: _fnc_remoteExecObjectJIPSafe
+    remote executes a function with a safe object JIP, The JIP key will be "FunctionName_ObjID"
+
+    Params:
```
Comment (killerswin2) [inline A3A/addons/core/functions/Save/fn_loadStat.sqf:35]: 😍

File: A3A/addons/core/functions/Save/fn_loadStat.sqf [hunk 2]
```diff
@@ -28,6 +28,26 @@ Example:
...
+        _function - [String] the function name to be remote executed (needs to be available in the global namespace of the targets)
+*/
+private _fnc_remoteExecObjectJIPSafe = {
+    params [
+         ["_object", objNull, [objNull]]
+        ,["_arguments", [], [[]]]
+        ,["_function", "", [""]]
+    ];
+
+    private _jipKey = _function+"_"+((str _object splitString ":") joinString "");
```
Comment (killerswin2) [inline A3A/addons/core/functions/Save/fn_loadStat.sqf:47]: nit pick: 
```suggestion
    private _jipKey = _function + "_" + ((str _object splitString ":") joinString "");
```

File: A3A/addons/core/functions/Save/fn_loadStat.sqf [hunk 3]
```diff
@@ -328,6 +348,29 @@ if (_varName in _specialVarLoads) then {
...
+                [_veh,[_veh],"A3A_fnc_initMovableObject"] call _fnc_remoteExecObjectJIPSafe;
+                [_veh,[_veh],"A3A_fnc_logistics_addLoadAction"] call _fnc_remoteExecObjectJIPSafe;
+                _veh allowDamage false;
+                _veh setVariable ["A3A_canGarage", true, true];
+                private _cat = if (typeOf _veh isEqualTo "vehicleFuelDrum") then {"vehicleFuelDrum"} else {"vehicleFuelTank"};
+                _veh setVariable ["A3A_itemPrice",
+                    FactionGet(reb,_cat)#1
+                , true];
+            };
+        /* we currently dont save the light sources (guessing because its a "building")
```
Comment (killerswin2) [inline A3A/addons/core/functions/Save/fn_loadStat.sqf:365]: Yes, light sources are "viewed" as temp buildings

File: A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf [hunk 4]
```diff
@@ -69,7 +69,7 @@ _item setVariable ["A3A_itemPrice", _price, true];
 {
     private _func_name = (_x #0);
     if (_x #1) then {
-            private _jipKey = "A3A_utilityItems_item_" + ((str _item splitString ":") joinString "");
+            private _jipKey = "A3A_utilityItems_item_"+_func_name+"_" + ((str _item splitString ":") joinString "");
```
Comment (killerswin2) [inline A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf:72]: another nit:
```suggestion
            private _jipKey = "A3A_utilityItems_item_" + _func_name + "_" + ((str _item splitString ":") joinString "");
```

File: A3A/addons/Garage/Public/fn_addVehicle.sqf [hunk 5]
```diff
@@ -49,32 +49,47 @@ if ([getPosATL _player] call A3A_fnc_enemyNearCheck) exitWith {
...
 
 //LTC refund
-if (_class in [FactionGet(occ,"surrenderCrate"), FactionGet(inv,"surrenderCrate")]) exitWith {
-    [_vehicle,boxX,true] call A3A_fnc_ammunitionTransfer;
-    [10] remoteExec ["A3A_fnc_resourcesPlayer", _client];
-    ["STR_HR_GRG_Feedback_addVehicle_LTC"] remoteExec ["HR_GRG_fnc_Hint", _client];
-    true
+private _ltcRefund = {
+    params ["_box", ["_instantRefund", true]];
+    [_box,boxX,true] call A3A_fnc_ammunitionTransfer;
```
Comment (killerswin2) [inline A3A/addons/Garage/Public/fn_addVehicle.sqf:54]: nit: 
```suggestion
    [_box, boxX, true] call A3A_fnc_ammunitionTransfer;
```

**Review by killerswin2 — APPROVED** [review]
Good work, I see no logic errors. Good to merge.

**Comment by jaj22** [comment]
The garaging stuff might be too quiet: Other than the refund, there's no indication of which objects were merged into arsenal/garage and players may not figure what happened to them, and so continue doing everything the slow/safe way. Would have been inclined to dump everything on the ground for that reason. Maybe the refund is sufficient though. 

The loadstat stuff looks correct but surely there's a lot of shared init code with buyItem there? Seems to be begging for a refactor. Also I think there's an issue that those JIPs are never cleaned up.

**Comment by killerswin2** [comment]
This also closes #2363

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Pull request merged**